kedro-org/kedro-viz
# Make Kedro-Viz shareable via a hosted URL

## Description

Closes #1116 

## Development notes

This PR enables Viz to be published and hosted on AWS S3.

For the frontend, most changes exist in `shareable-url-modal.js`. On the deployed version, we show a timestamp for context, which comes from the `shareable-url-metadata.js` file. 

A new feature hint for the Sharable URLs button is added in `feature-hints-content.js`. I've also updated `feature-hints.js` to remove that feature hint from the deployed version, since it doesn't make sense for the user to see it there.



## QA notes

> [!IMPORTANT]  
> You don't need to review any Python files in this PR (nothing inside `/package`)!!! That review will be done in #1498.

To get setup to test this locally:

- Install `fsspec` version `2023.9.0` or greater.
- Ensure you're running `kedro` version `18.2.0` or greater.
- You'll need access to an AWS S3 bucket and know the following information:
    - Access key
    - Secret access key
    - Bucket region
    - Bucket name
- Enable static website hosting on your S3 bucket and then make the objects public so you'll be able to view Viz when it's deployed 
- Set your access keys in your terminal, and then run `make run`.
- Then run the frontend with `npm start`.
- The deploy button is in the lower-left, between the theme toggle and settings icon.

## Checklist



<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1487"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
